jupyter-extension/jupyterlab_pachyderm/pfs_manager.py

```diff
@@ -479,24 +506,65 @@ def __init__(self, client: Client, **kwargs):
                 f"Could not create mount dir {PFS_MOUNT_DIR}. Try setting the PFS_MOUNT_DIR env var."
             )
             raise e
-        super().__init__(**kwargs)
 
-    # TODO: don't ignore name in the input spec
-    # right now, when a repo is mounted as part of mounting datum(s), the
-    # toplevel directory for the repo is project_repo_branch. we should
-    # update this to respect the name in the pfsinput objects being supplied
-    # to the mount_datums api. however, this is not that simple as the name
-    # can't simply be looked up in the datuminfo. we will need to have some
-    # mapping to track which files within a datum belong to which name.
+    def _populate_name_table(self, input: pps.Input):
+        """
+        Given the input spec, populates the bi-directional name to commit_uri
+        mapping of the commits mounted by the input.
+        """
+        if input.pfs:
+            repo = pfs.Repo(
+                name=input.pfs.repo,
+                project=pfs.Project(
+                    name=input.pfs.project if input.pfs.project else "default"
+                ),
+            )
+            branch = pfs.Branch(
+                repo=repo, name=input.pfs.branch if input.pfs.branch else "master"
+            )
+            commit_id = (
+                input.pfs.commit
+                if input.pfs.commit
+                else self._client.pfs.inspect_branch(branch=branch).head.id
```

Comment:
Should we catch if this errors? I could see a situation where the branch doesn't exist.

Reply:
yes, good point. i think throwing a ValueError would be better there. along that same vein we should be catching input errors and returning 400 status codes on them rather than 500s as we are doing now.